lib/minstall.js

```diff
@@ -34,48 +39,28 @@ function logIfInRoot(message) {
   }
 }
 
-function logVersionConflict(messages, moduleFolderName) {
-
-  const moduleFolder = path.join(cwd, moduletools.modulesFolder, moduleFolderName);
-  console.log(`
-
-| UNCRITICAL DEPENDENCY-CONFLICTS FOUND:
-|   ${messages.join('\n|   ')}
-|
-| This is not an error, but consider using compatible package-versions
-| throughout the whole project.`);
-}
-
 function getLocalPackageInfo() {
-  return ModuleInfo.loadFromFolder('', '');
-}
-
-function getInstalledModules() {
-  return moduletools.getModules('node_modules');
-}
-
-function removeModulesFromPackageList(packageNames, moduleInfos) {
-  const moduleFolders = moduleInfos.map((moduleInfo) => {
-    return moduleInfo.folderName;
-  });
-
-  return packageNames.filter((packageName) => {
-    if (moduleFolders.indexOf(packageName) >= 0) {
-      logger.debug(`skipping ${packageName}`);
-    }
-    return !(moduleFolders.indexOf(packageName) >= 0);
-  });
+  return ModuleInfo.loadFromFolder(cwd, '');
 }
 
 function checkStartConditions() {
   logger.debug('checking start conditions');
   return Promise.all([
     systools.verifyFolderName(cwd, 'node_modules'),
     getLocalPackageInfo(),
+    systools.runCommand('npm --version'),
   ])
     .then((results) => {
       const folderName = results[0];
       localPackage = results[1];
+      npmVersion = results[2];
+
+      // npm 5 workaround until npm-issue #16853 is fixed
+      if (semver.major(npmVersion) === 5) {
+        logger.info('npm 5 detected. forcing --cleanup');
+        cleanup = true;
+      }
```

Comment:
you could extract the check to a function, then initialize the cleanup variable at start and make it `const`.